addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeWork.java

```diff
@@ -0,0 +1,144 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.transcribe;
+
+import static org.nuxeo.ai.AIConstants.TRANSCRIBE_FACET;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.work.AbstractWork;
+import org.nuxeo.ecm.platform.video.TranscodedVideo;
+import org.nuxeo.ecm.platform.video.VideoDocument;
+import org.nuxeo.runtime.api.Framework;
+
+import software.amazon.awssdk.services.transcribestreaming.model.Alternative;
+import software.amazon.awssdk.services.transcribestreaming.model.LanguageCode;
+
+public class TranscribeWork extends AbstractWork {
+
+    private static final String CATEGORY = "transcribeWork";
+
+    public static final String TRANSCRIBE_RESULTS_PROP = "transcribe:results";
+
+    public static final String START_TIME_FIELD = "start";
+
+    public static final String END_TIME_FIELD = "end";
+
+    public static final String TYPE_FIELD = "type";
+
+    public static final String CONTENT_FIELD = "content";
+
+    public static final String TRANSCRIPT_FIELD = "transcript";
+
+    public static final String ITEMS_FIELD = "items";
+
+    public static final String DEFAULT_CONVERSION = "WAV 16K";
+
+    public static final LanguageCode DEFAULT_LANG_CODE = LanguageCode.EN_US;
+
+    protected final String TITLE;
+
+    public TranscribeWork(String repository, String docId) {
+        super(CATEGORY + "_" + docId);
+        this.repositoryName = repository;
+        this.docId = docId;
+        TITLE = CATEGORY + "_" + docId;
+    }
+
+    @Override
+    public void work() {
+        openSystemSession();
+        DocumentModel doc = session.getDocument(new IdRef(docId));
+
+        VideoDocument video = doc.getAdapter(VideoDocument.class);
+        if (video == null) {
+            throw new NuxeoException("Not a Video document; id = " + docId);
+        }
+
+        TranscodedVideo wav = video.getTranscodedVideo(DEFAULT_CONVERSION);
+        if (wav == null) {
+            throw new NuxeoException("No supported audio conversion found. Transcribe requires " + DEFAULT_CONVERSION);
+        }
+
+        if (!doc.hasFacet(TRANSCRIBE_FACET)) {
+            doc.addFacet(TRANSCRIBE_FACET);
+        }
+
+        TranscribeService service = Framework.getService(TranscribeService.class);
+        CompletableFuture<List<Alternative>> future = service.transcribe(wav.getBlob(), DEFAULT_LANG_CODE);
+        List<Alternative> result = future.join();
```

Comment:
is there any thing to take into account if the transcription is taking a long long time (for a movie of big size)? I don't see any transaction or session timeout issue here but did we try to do some tests with big assets to make sure?

Reply:
good catch. need to revisit